src/Screens/Products/Form/index.js

```diff
@@ -1,12 +1,162 @@
-import React from 'react'
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 import styles from './form.module.css';
+import Input from '../../../Components/Shared/Input';
+import { useForm } from "react-hook-form";
+
+const Form = (props) => {
+
+  const [formMode, setFormMode] = useState(true);
+  const [formText, setFormText] = useState('Add Product');
+  const params = useParams();
+  const id = params.id ? params.id : '';
+
+  const { register, setValue, handleSubmit, formState: { errors } } = useForm();
+
+  useEffect(() => {
+    async function fetchProduct() {
+      if (id) {
+        try {
+          const response = await fetch(`${process.env.REACT_APP_API_URL}/product/${id}`, {
+            method: 'GET'
+          });
+          const json = await response.json();
+          setFormMode(false);
+          setFormText('Update Product');
+          setValue("name", json.data.name);
+          setValue("description", json.data.description);
+          setValue("price", json.data.price['$numberDecimal']);
+          setValue("stock", json.data.stock);
+        } catch (error) {
+          alert('Could not GET Product.', error);
+        }
+      } else {
+        return null;
+      }
+    }
+    fetchProduct();
+  }, [id]);
+
+
+  const onSubmit = async (event) => {
+    if (formMode) {
+      try {
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/product/add`, {
```

Comment:
now that you have set redux move this logic into thunks

Reply:
this is already solved